include/grpc++/channel_interface.h

```diff
@@ -57,6 +58,23 @@ class ChannelInterface : public CallHook,
   virtual void* RegisterMethod(const char* method_name) = 0;
   virtual Call CreateCall(const RpcMethod& method, ClientContext* context,
                           CompletionQueue* cq) = 0;
+
+  // Get the current channel state. If the channel is in IDLE and try_to_connect
+  // is set to true, try to connect.
+  virtual grpc_connectivity_state GetState(bool try_to_connect) = 0;
+
+  // Return the tag on cq when the channel state is changed or deadline expires.
+  // GetState needs to called to get the current state.
+  template <typename T>
+  virtual void NotifyOnStateChange(grpc_connectivity_state last_observed,
+                                   const T& deadline,
+                                   CompletionQueue* cq, void* tag) = 0;
+
+  // Blocking wait for channel state change or deadline expires.
+  // GetState needs to called to get the current state.
+  template <typename T>
+  virtual bool WaitForStateChange(grpc_connectivity_state last_observed,
```

Comment:
This can be achieved using WaitForStateChange and GetState so that probably it is not worth adding it.